### DISCUSSION TITLE
Build form using database

### DISCUSSION THREAD

**Discussion by francoism90** [filamentphp/filament#5779] [Help] [answered]
I really like Filament Form, it works perfectly fine when building forms and registering them in a component. :)

However, would it be possible to somehow build a form/schema dynamically?
The reason being we need to allow users to make their own forms, e.g. like Google Forms.

One of the main reasons of concerns is the `rules`/validation aspect:
```php
<?php
// Just a simple example

$field = Field::first();

$form = TextInput::make($field->key)
    ->required($field->required)
    ->label($field->label)
    ->email($field->email)
    ->numeric($field->numeric)
    ..
```
We could also divide them by type instead: `email`, `phonenumber`, `numeric`, `textarea`, etc.

However, something like this would also be cool:
```php
$form = TextInput::make($field->key)
    ->when($field->required, fn (TextInput $input) => $input->required()),
    ->when($field->email, fn (TextInput $input) => $input->email()),
    ->when($field->tel, fn (TextInput $input) => $input->tel()),
    ..
```
However the `when` method does not seems to exists?

Do you have any ideas? The problem is not building the structure (database, views, etc.), it's the best way to handle this with Filament. :)

**Comment by danharrin** [comment, marked as answer]
Check out this demo - https://github.com/danharrin/form-builder

**Reply by francoism90** [reply]
@danharrin Thanks!

https://github.com/danharrin/form-builder/blob/main/app/Http/Livewire/Form.php#L31

However, how to pass more options/attributes?

**Reply by danharrin** [reply]
Just chain them on... `->email($config['is_email'])` etc. Every method accepts a conditional parameter

**Reply by francoism90** [reply]
Hmm, what about combining them?

```php
->email($config['is_email']) // false
->tel($config['is_tel']) // false
```
They are still populated to the rules, right?
Or are `false`/`null` properties ignored in the final `rules` object?

Also some don't seem to accept a `bool`, but maybe I'm mistaken? :)

Thanks!

**Reply by danharrin** [reply]
It should work as you expect, try it :)

**Reply by francoism90** [reply]
@danharrin Many thanks!
